test/scope.js

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*global describe,it*/
+
+const assert = require('insist');
+
+const Scope = require('../lib/scope');
+
+describe('Scope', function() {
+
+  describe('constructor', function() {
+
+    it('should accept a space-separated string', function() {
+      var s1 = Scope('a b c');
+      assert.deepEqual(s1.values(), ['a', 'b', 'c']);
+    });
+
+    it('should accept an array', function() {
+      var s1 = Scope(['a', 'b', 'c']);
+      assert.deepEqual(s1.values(), ['a', 'b', 'c']);
+    });
+
+    it('should accept a Scope instance', function() {
+      var s1 = Scope(['a', 'b', 'c']);
+      var s2 = Scope(s1);
+      assert.equal(s1, s2);
+    });
+
+  });
+
+  describe('has', function() {
+
+    it('should work with a single value', function() {
+      var s1 = Scope('foo bar');
+      assert(s1.has('foo'));
+      assert(s1.has('bar'));
+      assert(!s1.has('baz'));
+    });
+
+    it('should work with another Scope object', function() {
+      var s1 = Scope('foo bar');
+      var s2 = Scope('bar');
+      assert(s1.has(s2));
```

Comment:
Worth checking that `!s2.has(s1)` because it's a subset, and you must have all scopes in the set.